src/storage/async_page_fetcher.hpp

```diff
@@ -27,6 +27,25 @@
 
 namespace cublog
 {
+  /* generic task to be used with to call a engine function taking one argument - the thread context
+   * and returning a single result that is then passed back via the callback
+   *  - call func: takes as single argument the thread context and returns value
+   *  - callback func: takes as single argument the value returned by the call func
+   *
+   * TODO: error handling
+   */
+  template <typename CALL_FUNC, typename CALLBACK_FUNC>
```

Comment:
Questionable design, as it lacks the ability to relay error, should it ever occur on the called function.
It is better to change this to be more in line with the other existing tasks - ie: template-less.